scarlet/source.py

```diff
@@ -224,10 +252,32 @@ def init_multicomponent_source(sky_coord, frame, observation, bg_rms, flux_perce
     # renormalize morphs: initially Smax
     for k in range(K):
         morphs[k] /= morphs[k].max()
+        if np.all(morphs[k] <= 0):
+            msg = "Zero or negative morphology for component {} at y={}, x={}"
+            logger.warning(msg.format(k, *skycoords))
```

Comment:
Shouldn't this check be before you normalize `morphs[k]`? If all of the values are negative then dividing by the max will always give a positive result, so I'm not sure that this will ever be true.